src/components/RowComponent/RowComponent.tsx

```diff
@@ -0,0 +1,94 @@
+import { type RowComponentProps } from 'react-window';
+import type { Country } from '../../types/emissions.types';
+import './RowComponent.css';
+import { memo, useCallback, useMemo } from 'react';
+
+export const RowComponent = memo(function RowComponent({
+  index,
+  style,
+  countries,
+  year,
+  columns,
+  prevYear,
+  highlightUpdates,
+}: RowComponentProps<{
+  countries: Country[];
+  year: number | null;
+  columns: string[];
+  prevYear: number | null;
+  highlightUpdates: boolean;
+}>) {
+  const country = countries[index];
+
+  const { yearData, prevYearData, iso_code } = useMemo(() => {
+    const yearData =
+      country.data.find((info) => info.year === year) ||
+      country.data[country.data.length - 1];
+    const prevYearData = prevYear
+      ? country.data.find((info) => info.year === prevYear) ||
+        country.data[country.data.length - 1]
+      : null;
+
+    return { yearData, prevYearData, iso_code: country.iso_code };
```

Comment:
It is utils function that better to be move to the utills folder to keep component more clear